dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassFile.java

```diff
@@ -18,23 +18,24 @@
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
+import com.google.common.base.MoreObjects;
 import java.nio.file.Path;
 import java.util.Objects;
 
 /**
  * A tuple of a class name and a jar file to uniquely locate the class implementation in a class
  * path.
  */
-final class ClassAndJar {
+final class ClassFile {
```

Comment:
Better, though I still feel like we're missing something here. Not quite sure what though.

Reply:
IMO, Java is missing tuple types.

Reply:
go/nopair makes interestiong reading :-)

Reply:
That's nice. Thanks.